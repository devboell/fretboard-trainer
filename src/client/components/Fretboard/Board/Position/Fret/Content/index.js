import React from 'react'
import pt from 'prop-types'
import Wrapper from './Wrapper'


const Content = ({ note }) =>
  <Wrapper>
    {note}
  </Wrapper>

Content.propTypes = {
  note: pt.string.isRequired,
}

export default Content
