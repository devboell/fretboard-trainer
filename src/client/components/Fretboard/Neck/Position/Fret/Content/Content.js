import React from 'react'
import pt from 'prop-types'
import { isNil } from 'ramda'
import { Note } from 'tonal'
import Wrapper from './Wrapper'

const formatSharp = sh => sh // sh.replace('#', '\u266F')
const formatFlat = fl => fl // fl.replace('b', '\u266D')

const formatEnharmonics = ([sh, fl]) =>
  `${formatSharp(sh)}/${formatFlat(fl)}`

const formatNote = (note, showEnharmonics) => {
  const enharmonic = Note.enharmonic(note)
  const hasEnharmonic = note !== enharmonic

  return showEnharmonics && hasEnharmonic
    ? formatEnharmonics([note, enharmonic])
    : formatSharp(note)
}
/* the content width is a big problem, some options to exlore:
- format this way 'D#|Eb5'
- make ur own glyph
- showPcs/pitches option
- equal length frets
*/
const Content = ({ content, showEnharmonics }) => {
  const isNote = !isNil(Note.name(content))
  const note = isNote
    ? formatNote(content, showEnharmonics)
    : undefined

  return (
    <Wrapper>
      {note || content}
    </Wrapper>
  )
}


Content.propTypes = {
  content: pt.string.isRequired,
  showEnharmonics: pt.bool.isRequired,
}

export default Content
