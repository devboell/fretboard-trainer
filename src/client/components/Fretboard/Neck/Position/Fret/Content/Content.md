Content example:

```js
<Content
  content={'E2'}
  showEnharmonics={false}
/>
```