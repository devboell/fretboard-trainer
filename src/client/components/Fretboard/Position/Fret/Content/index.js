import React from 'react'
import pt from 'prop-types'
// import { isNil } from 'ramda'
// import { Note } from 'tonal'
import Wrapper from './Wrapper'

/*
const formatSharp = sh => sh.replace('#', '\u266F')
const formatFlat = fl => fl.replace('b', '\u266D')

const formatEnharmonics = ([sh, fl]) =>
  `${formatSharp(sh)}/${formatFlat(fl)}`

const formatNote = (note, showEnharmonic) => {
  const enharmonic = Note.enharmonic(note)
  const hasEnharmonic = note !== enharmonic

  return showEnharmonic && hasEnharmonic
    ? formatEnharmonics([note, enharmonic])
    : formatSharp(note)
}
*/

/* the content width is a big problem, some options to exlore:
- format this way 'D#/Eb5'
- make ur own glyph
- showPcs/pitches option
*/

const Content = ({ content }) =>
  /*
  const showEnharmonic = true
  const isNote = !isNil(Note.name(content))
  const note = isNote
    ? formatNote(content, showEnharmonic)
    : undefined

  {note || content}
  */

  (
    <Wrapper>
      {content}
    </Wrapper>
  )


Content.propTypes = {
  content: pt.string.isRequired,
}

export default Content
