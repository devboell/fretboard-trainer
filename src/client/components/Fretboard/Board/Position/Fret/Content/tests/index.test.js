import React from 'react'

import Content from '../index'

it('Content component, snapshot', () => {
  const wrapper = shallow(<Content note="E2" />)
  expect(wrapper).toMatchSnapshot()
})
