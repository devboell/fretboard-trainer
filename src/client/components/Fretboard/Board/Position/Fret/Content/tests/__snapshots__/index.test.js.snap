// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Content component, snapshot 1`] = `
<styled.div>
  E2
</styled.div>
`;
